src/transformStateWithClones.js

```diff
@@ -6,8 +6,27 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      newState = { ...newState };
+
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    } else if (action.type === 'clear') {
+      newState = {};
+    }
+    states.push({ ...newState });
+  }
```

Comment:
You know that you have only 3 action.type, so better use Switch construction on this situation. it makes the code more understandable

```
switch (action.type) {
      case 'addProperties':
        ...
        break;

      case 'removeProperties':
        ...
        break;

      case 'clear':
        ...
        break;
    }
```

Reply:
Fix this comment